plugins/collapsible-layers.js

```diff
@@ -56,22 +63,30 @@ function setup () {
           }
         }, this);
 
-        var name = this._name;
-        if (disabled[0]) {
-          name = '<i>' + name + '</i>';
+        if (disabled.length === this._allLayers.length) {
+          disabled.forEach(this.addLayer, this);
+          disabled.length = 0;
+          disabled.all = true;
+        }
+
+        var label;
+        if (disabled.length) {
           var enableAll = function (e) {
             e.preventDefault();
-            disabled.forEach(this._map.addLayer, this._map);
+            var map = e.control._map;
+            disabled.forEach(map.addLayer, map);
             // restore name
-            e.control._layerInfo(this).name = this._name;
-            e.control._update();
+            e.control.setLabel(this, this._name);
           };
           this.once('longclick', enableAll);
           this.once('expand', function () {
             this.off('longclick', enableAll);
           });
+          label =  this._name + ' [' + disabled.length + ']';
+          label = '<i>' + label + '</i>';
         }
-        this._ctrl.addOverlay(this, name);
+        this.options.defaultDisabled = disabled.all;
+        this._ctrl.addOverlay(this, label ||  this._name);
```

Comment:
```suggestion
        this._ctrl.addOverlay(this, label);
```

Reply:
I wrote exactly like you suggested first, but then changed for the sake of expression,

Reply:
in this case, I would separate calls with the `if` above (style anyway)